sda-download/api/api.go

```diff
@@ -65,12 +71,16 @@ func Setup() *http.Server {
 	}
 
 	router.HandleMethodNotAllowed = true
+	mw := SelectedMiddleware()
+
+	// protected endpoints
+	router.GET("/metadata/datasets", append(mw, sda.Datasets)...)
+	router.GET("/metadata/datasets/*dataset", append(mw, sda.Files)...)
+	router.GET("/files/:fileid", append(mw, sda.Download)...)
+	router.GET("/s3/*path", append(mw, s3.Download)...)
+	router.HEAD("/s3/*path", append(mw, s3.Download)...)
```

Comment:
```suggestion
	// protected endpoints
	router.GET("/metadata/datasets", append(SelectedMiddleware(), sda.Datasets)...)
	router.GET("/metadata/datasets/*dataset", append(SelectedMiddleware(), sda.Files)...)
	router.GET("/files/:fileid", append(SelectedMiddleware(), sda.Download)...)
	router.GET("/s3/*path", append(SelectedMiddleware(), s3.Download)...)
	router.HEAD("/s3/*path", append(SelectedMiddleware(), s3.Download)...)
```

Since the SelectedMiddleware function doesn't return any errors there is no need to generate a variable for this section.